lib/migrator.js

```diff
@@ -81,43 +106,36 @@ internals.symlinkMigrationFile = async (migrationFile, symlinkConfig) => {
         return;
     }
 
-    const file = Path.join(symlinkConfig.migrationsDir, migrationFile);
+    const file = Path.join(symlinkConfig.migrationDir, migrationFile);
     const link = Path.join(symlinkConfig.tmpPath, migrationFile);
 
-    await internals.PROMISIFIED_UTILITIES.symlinkAsync(file, link);
-};
-
-exports.migrate = async (knexGroups, defaultKnex, rollback) => {
-
-    const knexMigrations = new Map();
-
-    // Compile migration info per knex instance
-
-    for (let i = 0; i < knexGroups.length; ++i) {
-        const knexGroup = knexGroups[i];
-        const migrationsDir = knexGroup.migrationsDir;  // Already normalized to abs path
+    await Util.promisify(Fs.symlink)(file, link);
 
-        if (!migrationsDir) {
-            continue;
-        }
+};
 
-        const knex = knexGroup.knex || defaultKnex;
+/**
+ * A non-standard one :) Wraps the Tmp.dir method in a promise. We do this manually instead of using Util.promisify as we do with the other callback-style methods
+ * because Tmp.dir's callback takes 2 non-error parameters, but Promises fulfill with only a single value and Util.promisify handles this by
+ * fulfilling with only the first parameter (the path to the temporary directory), which resulted in us being unable to cleanup our temporary directory because we no longer had access to Tmp's cleanup method
+ *
+ * @param {object} opts — Options object that Tmp.dir normally expects
+ * @return {promise} Promise that, if Tmp.dir succeeds, fulfills with an object containing
+ *      - *tmpPath* the path to the new temp directory
+ *      - *cleanupCallback* {function} removes the temporary directory when we're done using it
+ */
+internals.promisifyTmpDir = (opts) => {
 
-        Hoek.assert(knex, 'Cannot specify a migrations directory without an available knex instance.');
+    return new Promise((fulfill, reject) => {
```

Comment:
Old school!  `fulfill` is better known today as `resolve`, though they are pretty much synonymous.  To do the "standard" thing, can we change this argument to `resolve`, though?

Reply:
Ahhhhh, gotcha. Wicked, thanks for pointing that out. For sure, I'll fix that 🙏